src/lib/core/CHIPConfig.h

```diff
@@ -776,9 +776,16 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
  * @def CHIP_IM_MAX_NUM_READ_HANDLER
  *
  * @brief Defines the maximum number of ReadHandler, limits the number of active read transactions on server.
+ *
+ * The default value comes from 3sub per fabric * max number of fabrics, then reserve 1 read client for each fabric.
+ *
+ * Note that there is one fabric for rotation slack, so the actual number of fabrics should be CHIP_CONFIG_MAX_FABRICS - 1.
+ *
+ * TODO: (#17085) Should be changed to (CHIP_CONFIG_MAX_FABRICS - 1) * 4 after we can hold more read handlers on more concise
+ * devices.
  */
 #ifndef CHIP_IM_MAX_NUM_READ_HANDLER
-#define CHIP_IM_MAX_NUM_READ_HANDLER 4
+#define CHIP_IM_MAX_NUM_READ_HANDLER ((CHIP_CONFIG_MAX_FABRICS - 1) * 3)
```

Comment:
Where are we doing the "reserve 1 read client for each fabric" thing?

Reply:
The logic is placed in InterationModelEngine, and there are static asserts to guarantee this.